### ISSUE TITLE
Schema refactor: collapse EQ sampler's two label columns into one nullable `boolean_value`

### ISSUE THREAD

**Issue by mmcdermott** [payalchandak/EveryQuery#122]
## ⚠️ Scope correction (2026-04-19)

Original version of this issue proposed dropping the censor prediction head and renaming `query` to `code`.  **Both were wrong** per clarification on #96.  Rewritten below.

## Motivation

The sampler's output currently carries two label columns:

- `boolean_value` — overloaded to mean *censored* (subject data ended before we could observe the outcome).
- `occurs` — the real "did the event fire" label.

Readers of MEDS-shaped parquets reasonably expect `boolean_value` to mean *the label value*, not *a censoring flag*.  I hit this the hard way writing the training-validity test in #119 — there's a big inline comment in `tests/test_training_validity.py` specifically documenting the gotcha.

But the sampler's two columns aren't independent: `occurs` is only meaningful **conditional on** `boolean_value` (the censoring flag) being `False`.  When censored, the `occurs` value is don't-care.  That structural conditioning is exactly what a **nullable** `boolean_value` expresses idiomatically in MEDS's `LabelSchema`:

| `boolean_value` | Meaning |
|---|---|
| `null` | Censored — we can't answer the query for this patient |
| `True` | Not censored AND event occurred in the window |
| `False` | Not censored AND event did not occur |

One column, three states, semantically aligned with MEDS conventions.

### Model code: **unchanged**

This is the key correction vs. the original issue.  The existing loss wiring in `model/model.py:614-621`

```python
censor_loss = self._get_loss(censor_logits, batch.censor, mask=None)
occurs_loss = self._get_loss(occurs_logits, batch.occurs, mask=~batch.censor)
```

keeps working as-is.  Both prediction heads stay — `censor` predicts "can we answer?" and `occurs` predicts "conditioned on being able to answer, what would the answer be?"  Both are meaningful learning targets.  Only the **source** of `batch.censor` / `batch.occurs` changes (one column → two derived tensors instead of two columns → two directly-copied tensors).

**Checkpoints stay loadable.**  `state_dict` keys are unchanged.  No backward-compat break for existing trained `.ckpt` files.

### Sampler (`evaluate_index_df`): single-column output

Current:
```python
.with_columns(
    pl.col("_censored").alias("boolean_value"),
    (pl.col("_censored").not_() & event_in_window).alias("occurs"),
)
.select("subject_id", "prediction_time", "boolean_value", "occurs", "query", "duration_days")
```

Target:
```python
.with_columns(
    pl.when(pl.col("_censored"))
    .then(None)
    .otherwise(event_in_window)
    .alias("boolean_value"),
)
.select("subject_id", "prediction_time", "boolean_value", "query", "duration_days")
```

No more `occurs` column in the emitted parquet.

### Dataset (`data/dataset.py::collate`): derive two tensors from one column

Current (paraphrased):
```python
out["censor"] = out[self.LABEL_COL]  # LABEL_COL == "boolean_value"
out["occurs"] = torch.Tensor([item["occurs"] for item in batch]).long()
```

Target:
```python
is_censored = out["boolean_value"].isnan()   # or equivalent null check for the dtype
out["censor"] = is_censored
out["occurs"] = out["boolean_value"].fill_null(False).long()  # placeholder where censored
```

Model's loss mask (`~batch.censor`) still zeros out the placeholder rows correctly.

### Not renaming `query` → `code`

Keeping `query` as the sampler's column name.  The future plan is to generalise `query` into a struct (ANY/ALL compound queries, event-bounded durations) — using `code` now would imply a flat-string contract that doesn't hold long-term.

**`TaskQuerySchema` in #96 needs its column renamed from `code` → `query`** to match.  That's a one-line change in `src/every_query/data/schema.py`.

## Affected call sites

| Component | Change |
|---|---|
| `data/schema.py::TaskQuerySchema` | `code` field renamed to `query` |
| `generate_tasks/sample_tasks.py::evaluate_index_df` | Emit nullable `boolean_value`; drop `occurs` column |
| `data/dataset.py::EveryQueryPytorchDataset.collate` | Derive `batch.censor` + `batch.occurs` from `boolean_value` |
| `data/dataset.py::LABEL_TENSOR_NAMES` | Drop `occurs` from the tuple |
| `model/model.py` | **Unchanged** |
| `tests/test_sample_tasks.py::TestEvaluateIndexDfEdgeCases` | Update expected `(boolean_value=None\|True\|False)` semantics |
| `tests/test_generate_tasks.py::test_labels_match_ground_truth` | Recompute single `boolean_value` per the new semantics |
| `tests/test_training_validity.py` | Synthetic labels: `boolean_value=<occurs_truth>` directly (no subjects are censored in this test); drop the inline gotcha comment |

## Risks

1. **Loss-mask regression** is still the biggest one — if `batch.censor` is derived incorrectly from `boolean_value`, `mask=~batch.censor` ends up wrong and training silently diverges.  #119 is the strongest existing guard (HOT-COLD AUROC drops to chance if the mask inverts or goes uniformly True).
2. **Paper-experiment scripts** may still reference the `occurs` column directly.  Audit `paper_experiments/` before the sampler change.  (Post-grep confirms nothing in `paper_experiments/` touches `occurs` today.)
3. **Checkpoint backward-compat**: preserved by keeping both heads.  No action needed.

## Landing plan

Bundle the migration into #96 itself (per owner preference — see #96 review thread).  Sequencing:

1. Merge **#119** first (on current semantics) so dev has the training-validity safety net before the migration commit.
2. Expand **#96** to include the full migration above.
3. Rebase #112, #113, #119's small test-only delta onto the expanded #96.
4. Merge expanded #96.

cc @mmcdermott @payalchandak

**Comment by mmcdermott** [comment]
Shipped in #96 — sampler now emits a single nullable `boolean_value` column (null = censored, True = event occurred, False = no event and not censored) instead of the old (boolean_value, occurs) pair.  The collapse + its inverse live in `EveryQueryPytorchDataset.__init__` (collapse) and `predict.py::_identifiers_from_schema_df` (un-collapse on output).  Closing on merge-to-dev.
